### PR TITLE
Add TabNet Datasets

### PR DESCRIPTION
**Pull request by kanishk16**
This PR updates the `KaggleDownloadMixin` and adds the following datasets from the TabNet paper:

- [x] Poker Hand 
- [ ] Rossmann Store Sales
- [x] Sarcos
- [ ] Adult Census Income 
- [x] Mushroom Edibility
- [ ] Forest cover
- [ ] KDD 2009 Churn
- [ ] KDD 2009 Upsell
- [ ] KDD 2009 Appetency

### CODE REVIEW

File: ludwig/datasets/mixins/process.py [hunk 1]
```diff
@@ -46,6 +46,9 @@ def read_file(self, filetype, filename, header=0):
         elif filetype == 'tsv':
             file_df = pd.read_table(
                 os.path.join(self.raw_dataset_path, filename))
+        elif filetype == 'data':
```
Comment (w4nderlust) [inline ludwig/datasets/mixins/process.py:49]: not sure, what's the difference between this and csv?

File: ludwig/datasets/sarcos/config.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+version: 1.0
+download_urls:
+  - "http://www.gaussianprocess.org/gpml/data/sarcos_inv.mat"
+  - "http://www.gaussianprocess.org/gpml/data/sarcos_inv_test.mat"
+split_filenames:
+  train_file: sarcos_inv.mat
+  test_file: sarcos_inv_test.mat
+download_file_type: mat
+csv_filename: poker_hand.csv
```
Comment (w4nderlust) [inline ludwig/datasets/sarcos/config.yaml:9]: this should be sarcos.csv

File: ludwig/datasets/mixins/load.py [hunk 3]
```diff
@@ -48,7 +48,12 @@ def load_processed_dataset(self, split) -> Union[pd.DataFrame,
                 training_set = data_df[data_df[SPLIT] == 0]
                 val_set = data_df[data_df[SPLIT] == 1]
                 test_set = data_df[data_df[SPLIT] == 2]
-                return training_set, test_set, val_set
+                if val_set.shape[0] == 0:
```
Comment (w4nderlust) [inline ludwig/datasets/mixins/load.py:51]: I would always return 3 elements for consistency. in case `len(*_set) == 0` I would replace it with `None` that is easy to check for.

File: ludwig/datasets/mixins/kaggle.py [hunk 4]
```diff
@@ -28,7 +28,7 @@ def download_raw_dataset(self):
         kaggle.json file we lookup the passed in username and the api key and
         perform authentication.
         """
-        with self.update_env(KAGGLE_USERNAME=self.kaggle_username, KAGGLE_API_KEY=self.kaggle_api_key):
+        with self.update_env(KAGGLE_USERNAME=self.kaggle_username, KAGGLE_KEY=self.kaggle_key):
```
Comment (w4nderlust) [inline ludwig/datasets/mixins/kaggle.py:31]: I would revert this, let's keep on using the API key and deal with tests differently

File: ludwig/datasets/goemotions/__init__.py [hunk 5]
```diff
@@ -20,7 +20,7 @@
 from ludwig.datasets.mixins.process import *
 
 
-def load(cache_dir=DEFAULT_CACHE_LOCATION, split=False):
+def load(cache_dir=DEFAULT_CACHE_LOCATION, split=True):
```
Comment (w4nderlust) [inline ludwig/datasets/goemotions/__init__.py:23]: why this change?

**Review by w4nderlust — CHANGES_REQUESTED** [review]

File: ludwig/datasets/goemotions/__init__.py (hunk 5 above)
Comment (kanishk16) [reply ludwig/datasets/goemotions/__init__.py:23]: Since it's config file contains separate train, val and test data

File: ludwig/datasets/mixins/process.py (hunk 1 above)
Comment (kanishk16) [reply ludwig/datasets/mixins/process.py:49]: Oh yes, I'll combine this with csv

File: ludwig/datasets/mixins/kaggle.py (hunk 4 above)
Comment (kanishk16) [reply ludwig/datasets/mixins/kaggle.py:31]: I have adapted the test that earlier failed and should probably work now. Since kaggle no longer recommends using `KAGGLE_API_KEY`, making this change to `KAGGLE_KEY` significant (c.f: https://github.com/Kaggle/kaggle-api#api-credentials)

File: ludwig/datasets/goemotions/__init__.py [hunk 6]
```diff
@@ -37,9 +37,9 @@ class GoEmotions(UncompressedFileDownloadMixin, MultifileJoinProcessMixin,
     def __init__(self, cache_dir=DEFAULT_CACHE_LOCATION):
         super().__init__(dataset_name="goemotions", cache_dir=cache_dir)
 
-    def read_file(self, filetype, filename):
+    def read_file(self, filetype, filename, header=None):
```
Comment (w4nderlust) [inline ludwig/datasets/goemotions/__init__.py:40]: why do w need to expose this parameter?

**Review by w4nderlust — APPROVED** [review]
